CondFormats/CTPPSReadoutObjects/src/CTPPSPixelGainCalibration.cc

```diff
@@ -0,0 +1,193 @@
+      << detid << " size peds = " << sensorSize << " size gains = " << gainsSize; 
+  DetRegistry myreg;
+  myreg.detid=detid;
+  myreg.ibegin=0;
+  myreg.iend=sensorSize;
+  myreg.ncols=sensorSize/160; // each ROC is made of 80 rows and 52 columns, each sensor is made of 160 rows and either 104 or 156 columns (2x2 or 2x3 ROCs)
+  indexes = myreg;
+  for(int i = 0 ; i<sensorSize ; ++i)  
+    putData(i,peds[i],gains[i]);
+}
+
+void CTPPSPixelGainCalibration::putData(uint32_t ipix, float ped, float gain){
+  if (v_pedestals.size()>ipix) //if the vector is too big, this pixel has already been set
+    {
+      if (ped>=0 && gain>=0)
+	throw cms::Exception("CTPPSPixelGainCalibration fill error")
+	  << "[CTPPSPixelGainCalibration::putData] attempting to fill the vectors that are already filled detid = " << indexes.detid << " ipix " << ipix;
+      else   // in case it is for setting the noisy or dead flag of already filled pixel you are allowed to reset it
+	{
+	  edm::LogInfo("CTPPSPixelGainCalibration")<<"resetting pixel calibration for noisy or dead flag";
+	  resetPixData(ipix,ped,gain);
+	}
+    }
+  else if (v_pedestals.size()<ipix)
+    throw cms::Exception("CTPPSPixelGainCalibration fill error")
+      << "[CTPPSPixelGainCalibration::putData] the order got scrambled detid = "<< indexes.detid << " ipix " << ipix;
+  else{ //the size has to match exactly the index, the index - 0 pixel starts the vector, the one with index 1 should be pushed back in a vector of size== 1 (to become size==2) so on and o forth
+    v_pedestals.push_back(ped);
+    v_gains.push_back(gain);
+  }
+}
+
+void CTPPSPixelGainCalibration::setIndexes(const uint32_t &detid){
+  indexes.detid  = detid;
+  indexes.ibegin = 0 ;
+  indexes.iend   = v_pedestals.size();
+  indexes.ncols  = indexes.iend/ 160 ; // nrows is 160 in CTPPS
+}
+
+void CTPPSPixelGainCalibration::resetPixData(uint32_t ipix, float ped, float gain){
+  if (v_pedestals.size()<=ipix){
+    edm::LogError("CTPPSPixelGainCalibration")<<"this element ipix = " << ipix << " has not been set yet"; 
+    return;
+  }
+  v_pedestals[ipix]=ped; //modify value to already exisiting element
+  v_gains[ipix]=gain;
+}
+
+float CTPPSPixelGainCalibration::getPed(const int& col, const int& row ) const {
+  int nCols=indexes.ncols;
+  int nRows=v_pedestals.size()/nCols;
+  if (col >= nCols || row >= nRows){
+    throw cms::Exception("CorruptedData")
+      << "[CTPPSPixelGainCalibration::getPed] Pixel out of range: col " << col << " row " << row;
+  }
+  bool  isDead = false;
+  bool  isNoisy= false;
+  int   ipix = col + row * nCols;
+  float tmpped  = getPed(ipix,isDead,isNoisy);
+  if (isDead || isNoisy)
+    edm::LogWarning("CTPPSPixelGainCalibration")<<"Using noisy or dead pixel";
+  return tmpped;
+}
+
+float CTPPSPixelGainCalibration::getPed(const int& col, const int& row, bool& isDead, bool& isNoisy) const {
+  int nCols=indexes.ncols;
+  int nRows=v_pedestals.size()/nCols;
```

Comment:
Actually I did, but I save the size which should be ncols*nrows so I thought it was implicit and desisted. Perhaps I should use that here then, thanks for the suggestion.